File: app/src/main/java/com/example/petridelivery/abs/BaseActivity.java
package com.example.petridelivery.abs;

import android.content.pm.ActivityInfo;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.petridelivery.app.PetriDeliveryApp;
import com.example.petridelivery.concurent.DaggerConcurrentComponent;
import com.example.petridelivery.concurent.TaskRunner;
import com.example.petridelivery.util.InjectionUtils;
import com.example.petridelivery.wrappers.WrapperComponent;

import lombok.Getter;

@Getter
public abstract class BaseActivity extends AppCompatActivity {

    protected WrapperComponent wc;
    protected PetriDeliveryApp app;
    protected TaskRunner runner = DaggerConcurrentComponent.create().getTaskRunner();

    @Override
    protected void onCreate(Bundle savedInstance){
        super.onCreate(savedInstance);
        setContentView(getLayoutId());

        app = (PetriDeliveryApp) getApplication();
        wc = InjectionUtils.getWrapperComponent(app);

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
    }

    protected abstract int getLayoutId();
}

File: app/src/main/java/com/example/petridelivery/abs/BaseActivity.kt
package com.example.petridelivery.abs

import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.petridelivery.app.PetriDeliveryApp
import com.example.petridelivery.util.getWrapperComponent
import com.example.petridelivery.wrappers.WrapperComponent


abstract class BaseActivity : AppCompatActivity() {
    lateinit var wc: WrapperComponent
    lateinit var app: PetriDeliveryApp

    override fun onCreate(savedInstance: Bundle?) {
        super.onCreate(savedInstance)
        setContentView(getLayoutId())

        app = application as PetriDeliveryApp
        wc = getWrapperComponent(app)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
    }

    protected abstract fun getLayoutId(): Int

}